### PR TITLE
async await - add expectation tests

### PR DESCRIPTION
**Pull request by KutnerUri**

### CODE REVIEW

File: test/samples/async.js [hunk 1]
```diff
@@ -138,5 +138,42 @@ module.exports = [
 		options: { transforms: { asyncAwait: true, dangerousForOf: true } },
 		input: `for await (const x of someFunction()) { x() }`,
 		error: /Transforming for-await-of statements is not implemented/
+	},
+
+	// variables interaction with async :
+	{
+		description: 'transpiles a variable decleration with await',
+		options: { transforms: { asyncAwait: true } },
```
Comment (adrianheine) [inline test/samples/async.js]: `asyncAwait` is enabled by default, so you don't have to pass it.

File: test/samples/async.js [hunk 2]
```diff
@@ -138,5 +138,42 @@ module.exports = [
 		options: { transforms: { asyncAwait: true, dangerousForOf: true } },
 		input: `for await (const x of someFunction()) { x() }`,
 		error: /Transforming for-await-of statements is not implemented/
+	},
+
+	// variables interaction with async :
+	{
+		description: 'transpiles a variable decleration with await',
```
Comment (adrianheine) [inline test/samples/async.js]:
```suggestion
		description: 'transpiles a variable declaration with await',
```

File: test/samples/async.js [hunk 3]
```diff
@@ -138,5 +138,42 @@ module.exports = [
...
+		options: { transforms: { asyncAwait: true } },
+		input: `async function f() { const a = await f1(); return a + 1; }`,
+		output: `function f() { return Promise.resolve().then(function(){ return f1(); }).then(function(a){ return a + 1;})`
+	},
+
+	{
+		description: 'transpiles multiple variable declerations in a row',
```
Comment (adrianheine) [inline test/samples/async.js]:
```suggestion
		description: 'transpiles multiple variable declarations in a row',
```

File: test/samples/async.js [hunk 4]
```diff
@@ -138,5 +138,42 @@ module.exports = [
...
+		options: { transforms: { asyncAwait: true } },
+		input: `async function f() { const a1 = await f1(); const a2 = await f2(); return a1 + a2; }`,
+		output: `function f() { const a1; const a2; return Promise.resolve().then(function(){ return f1(); }).then(function(ref){ a1 = ref; return f2(); }).then(function(ref) { a2 = ref; return a1 + a2; }); }`
```
Comment (adrianheine) [inline test/samples/async.js]: You cannot declare `const` variables without initializing them, so you should use `var` instead.

File: test/samples/async.js [hunk 5]
```diff
@@ -19,7 +19,7 @@ module.exports = [
 	{
 		description: 'errors on async function declarations if transform is enabled',
 		input: `async function A() {}`,
-		error: /Transforming async functions is not implemented/
+		output: `function A() { return Promise.resolve(); }`
```
Comment (adrianheine) [inline test/samples/async.js]: I think you can leave the `error` and add `output`. That way, the tests still pass until we remove the `error` property.

**Comment by adrianheine** [comment]
Thanks!

**Comment by KutnerUri** [comment]
stale :(

**Comment by adrianheine** [comment]
I did provide specific feedback, though.

**Pull request closed without merging**